base_layer/wallet/src/transaction_service/protocols/transaction_coinbase_monitoring_protocol.rs

```diff
@@ -154,275 +154,427 @@ where TBackend: TransactionBackend + 'static
+                Some(c) => c,
+            };
+            let mut client = match base_node_connection
+                .connect_rpc_using_builder(
+                    BaseNodeWalletRpcClient::builder().with_deadline(self.resources.config.chain_monitoring_timeout),
+                )
+                .await
+            {
+                Ok(c) => c,
+                Err(e) => {
+                    warn!(
+                        target: LOG_TARGET,
+                        "Problem establishing RPC connection (TxId: {}): {}", self.tx_id, e
+                    );
+                    delay.await;
+                    continue;
+                },
+            };
+
+            let signature = completed_tx.transaction.body.kernels()[0].clone().excess_sig;
+            let delay = delay_for(self.timeout).fuse();
+            // Loop until both a Mempool response AND a Base node response is received OR the Timeout expires.
+            loop {
+                futures::select! {
+                    new_base_node = base_node_update_receiver.select_next_some() => {
+                        match new_base_node {
+                            Ok(bn) => {
+                                self.base_node_public_key = bn;
+                                 info!(
                                     target: LOG_TARGET,
-                                    "Failed to Cancel outputs for Coinbase TX_ID: {} with error: {:?}",
-                                    completed_tx.tx_id,
-                                    e
+                                    "Coinbase Monitoring protocol (TxId: {}) Base Node Public key updated to {:?}",
+                                    self.tx_id,
+                                    self.base_node_public_key
                                 );
-                            }
-                            if let Err(e) = self.resources.db.cancel_completed_transaction(completed_tx.tx_id).await {
-                                warn!(
+                                self.first_rejection = false;
+                                continue;
+                            },
+                            Err(e) => {
+                                trace!(
                                     target: LOG_TARGET,
-                                    "Failed to Cancel Coinbase TX_ID: {} with error: {:?}", completed_tx.tx_id, e
+                                    "Coinbase Monitoring protocol (TxId: {}) event 'base_node_update' triggered with \
+                                     error: {:?}",
+                                    self.tx_id,
+                                    e,
                                 );
                             }
+                        }
+                    }
+                    result = self.query_coinbase_transaction(signature.clone(), &mut client).fuse() => {
+                        let (utxo_found, meta_data) = match result {
+                            Ok(r) => (r.0, r.1),
```

Comment:
You can either just destructure the tuple here, or just return it: 

`Ok((utxo_found, metadata)) => (utxo_found, metadata),` or 

`Ok(response) => response,`

Reply:
Implemented the 2nd option, thanks.
  _(I do not think the first option improves the code)_